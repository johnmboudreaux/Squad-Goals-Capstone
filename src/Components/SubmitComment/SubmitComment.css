.submit-comment {
  width: 100%;
  margin: auto;
  display: flex;
  margin-top: 10px; }

.submit-comment-form {
  width: 100%; }

.submit-comment input {
  height: 40px; }

.submit-comment-input {
  width: 75%;
  min-width: 250px; }

.submit-comment-button {
  width: 20%;
  min-width: 75px; }
