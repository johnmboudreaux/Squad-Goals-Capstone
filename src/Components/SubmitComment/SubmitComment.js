import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SubmitComment extends Component {
  // Need to get the conversation_id via props and need to rework
  // the server endpoint to post a comment to just take a conversation_id

  postNewComment() {
    const commentBody = document.querySelector('.squad-name').value;
    const postBody = {
      body: commentBody,
      conversation_id: this.props.conversation_id,
    };
    fetch('/api/v1/comments', {
      headers: {
        'Content-Type': 'application/json',
      },
      method: 'POST',
      body: JSON.stringify(postBody),
    })
      .then(response => console.log(response))
      .catch(error => console.error(error));
  }

  render() {
    return (
      <div className="submit-comment">
        <div className="submit-comment-form">
          <input
            className="submit-comment-input"
            type="text"
            placeholder="What do you want to say?"
          />
          <input
            onClick={() => {
              this.postNewComment();
            }}
            type="button"
            value="Submit"
          />
        </div>
      </div>
    );
  }
}

SubmitComment.propTypes = {
  conversation_id: PropTypes.number,
};

export default SubmitComment;
