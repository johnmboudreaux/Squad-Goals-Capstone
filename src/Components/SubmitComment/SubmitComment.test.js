import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import SubmitComment from './SubmitComment';
// import goal from '../../../data/test/goals_data_test';

describe('SubmitComment snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<SubmitComment />);

    expect(wrapper).toMatchSnapshot();
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SubmitComment />, div);
  });
});
