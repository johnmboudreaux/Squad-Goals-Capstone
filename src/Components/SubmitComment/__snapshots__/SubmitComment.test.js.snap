// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitComment snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitComment />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="submit-comment-form"
      >
        <input
          className="submit-comment-input"
          placeholder="What do you want to say?"
          type="text"
        />
        <input
          onClick={[Function]}
          type="button"
          value="Submit"
        />
      </div>,
      "className": "submit-comment",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="submit-comment-input"
            placeholder="What do you want to say?"
            type="text"
          />,
          <input
            onClick={[Function]}
            type="button"
            value="Submit"
          />,
        ],
        "className": "submit-comment-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "submit-comment-input",
            "placeholder": "What do you want to say?",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
            "type": "button",
            "value": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="submit-comment-form"
        >
          <input
            className="submit-comment-input"
            placeholder="What do you want to say?"
            type="text"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Submit"
          />
        </div>,
        "className": "submit-comment",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="submit-comment-input"
              placeholder="What do you want to say?"
              type="text"
            />,
            <input
              onClick={[Function]}
              type="button"
              value="Submit"
            />,
          ],
          "className": "submit-comment-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "submit-comment-input",
              "placeholder": "What do you want to say?",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "button",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
